File: ForRadiant/JoinPanelID/Form1.vb
Imports System.Reflection
Public Class Form1
    Private Sub SetVersionInfo()

        Dim ass As System.Reflection.Assembly = System.Reflection.Assembly.GetExecutingAssembly()
        Dim ver As System.Version = ass.GetName().Version
        Dim startDate As DateTime = New Date(2000, 1, 1)
        Dim diffDays As Integer = ver.Build
        Dim computedDate As DateTime = startDate.AddDays(diffDays)
        Dim lastBuilt As String = computedDate.ToShortDateString()
        'Me.Text = (Me.Text & " " & ver.Major & "." & ver.Minor & "." & ver.Build & "." & ver.Revision & " (" & lastBuilt & ")")
        Me.Text = (Me.Text & " " & ver.Major & "." & ver.Minor & "." & ver.Build & "." & ver.Revision)

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If Not TextBox1.Text = "" And Not TextBox2.Text = "" Then
            TextBox4.Text = ""

            For Each s As String In Me.TextBox1.Text.Split({Environment.NewLine}, StringSplitOptions.None)

                If Not s = "" Then
                    If s.Length > TextBox2.Text Then
                        s = s.Substring(0, TextBox2.Text)
                        TextBox4.Text += s & TextBox3.Text
                    Else
                        TextBox4.Text += s & TextBox3.Text
                    End If
                End If

            Next

            TextBox4.Text = TextBox4.Text.Substring(0, TextBox4.Text.Length - TextBox3.TextLength)
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Clipboard.Clear()
        Clipboard.SetText(TextBox4.Text)
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        SetVersionInfo()
    End Sub
End Class

File: ForRadiant/FTPIndexPathGenerator/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Runtime.InteropServices.WindowsRuntime;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace FTPIndexPathGenerator
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void SetVersionInfo()
        {
            Version versionInfo = Assembly.GetExecutingAssembly().GetName().Version;
            DateTime startDate = new DateTime(2000, 1, 1);
            int diffDays = versionInfo.Build;
            DateTime computedDate = startDate.AddDays(diffDays);
            string lastBuilt = computedDate.ToShortDateString();
            //this.Text = string.Format("{0} - {1} ({2})",
            //            this.Text, versionInfo.ToString(), lastBuilt);
            this.Text = string.Format("{0} - {1}",
                        this.Text, versionInfo.ToString());
        }
        private void button1_Click(object sender, EventArgs e)
        {
            int nSeed = 7919; // nSeed = 7919
            int LayerSize;
            int LayerCount = 1;
            string PID = textBox1.Text;
            int nVlue = 0;
            long lTmp;
            string path;
            char[] cPanlID = PID.ToCharArray();

            // Hash For higher directory -> num = 0
            // Hash For lower directory -> num = 1
            //int num = 0;

            for (int i = 0; i < PID.Length; i++)
            {
                char ctemp = cPanlID[i];
                int num = 0x00ff;
                int temp = 0x00ff & cPanlID[i];
                nVlue = ((nVlue * 0x00ff) + (0x00ff & cPanlID[i])) % nSeed;
            }

            if (nSeed <= 157)
            {
                LayerSize = nSeed;
            }
            else
            {
                LayerCount = 2;
                LayerSize = Convert.ToInt32(nSeed / Math.Pow(nSeed, 0.5));
            }

            
            if (LayerCount == 1)
            {
                lTmp = Convert.ToInt32(nVlue / LayerSize - 0.49999);
                path = lTmp.ToString("00000000");
            }
            else
            {
                //DongA logic
                //double dTemp;
                //dTemp = nVlue / LayerSize - 0.49999 ;
                //path = dTemp.ToString("00000000");

                lTmp = Convert.ToInt32(nVlue / LayerSize - 0.49999);
                path = lTmp.ToString("00000000");
                lTmp = nVlue % LayerSize;
                path = path + "\\" + lTmp.ToString("00000000");
            }

            textBox2.Text = path;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            SetVersionInfo();
        }
    }
}
